src/mdns/mdns_avahi.cpp

```diff
@@ -718,38 +718,43 @@ void PublisherAvahi::UnpublishService(const std::string &aName, const std::strin
     std::move(aCallback)(error);
 }
 
-void PublisherAvahi::PublishHostImpl(const std::string &         aName,
-                                     const std::vector<uint8_t> &aAddress,
-                                     ResultCallback &&           aCallback)
+void PublisherAvahi::PublishHostImpl(const std::string &            aName,
+                                     const std::vector<Ip6Address> &aAddresses,
+                                     ResultCallback &&              aCallback)
 {
     otbrError        error      = OTBR_ERROR_NONE;
     int              avahiError = AVAHI_OK;
     std::string      fullHostName;
-    AvahiAddress     address;
     AvahiEntryGroup *group = nullptr;
 
     VerifyOrExit(mState == State::kReady, error = OTBR_ERROR_INVALID_STATE);
     VerifyOrExit(mClient != nullptr, error = OTBR_ERROR_INVALID_STATE);
-    VerifyOrExit(aAddress.size() == sizeof(address.data.ipv6.address), error = OTBR_ERROR_INVALID_ARGS);
 
-    aCallback = HandleDuplicateHostRegistration(aName, aAddress, std::move(aCallback));
+    aCallback = HandleDuplicateHostRegistration(aName, aAddresses, std::move(aCallback));
     VerifyOrExit(!aCallback.IsNull());
+    VerifyOrExit(!aAddresses.empty(), std::move(aCallback)(OTBR_ERROR_NONE));
+
+    VerifyOrExit((group = CreateGroup(mClient)) != nullptr, error = OTBR_ERROR_MDNS);
 
-    address.proto = AVAHI_PROTO_INET6;
-    memcpy(address.data.ipv6.address, aAddress.data(), aAddress.size());
     fullHostName = MakeFullHostName(aName);
+    for (const auto &address : aAddresses)
+    {
+        AvahiAddress avahiAddress;
+
+        avahiAddress.proto = AVAHI_PROTO_INET6;
+        memcpy(avahiAddress.data.ipv6.address, address.m8, sizeof(address.m8));
+        avahiError = avahi_entry_group_add_address(group, AVAHI_IF_UNSPEC, AVAHI_PROTO_UNSPEC, AVAHI_PUBLISH_NO_REVERSE,
```

Comment:
Are we ignoring avahi errors intentionally?

Reply:
Good catch! I misplaced the `VerifyOrExit`.

Reply:
From another perspective, I wonder if we should continue committing the group as long as one address is successfully registered. 
Not sure if it could happen at all in practise.

Reply:
I took a look at `avahi/avahi-utils/avahi-publish.c` and I think this case is unlikely to happen.